File: 1.22/timed_prime_test.rs
#[path= "../1.21/smallest_divisor.rs"]
mod smallest_divisor;

use std::time::{Duration, Instant};

pub fn search_for_primes(n: i64, count: i64) {
    if is_prime(n) {
        report(n);
        if count > 1 {
            search_for_primes(n + 1, count - 1);
        }
    } else {
        search_for_primes(n + 1, count);
    }
}

fn is_prime(n: i64) -> bool {
    return n == smallest_divisor::calc(n);
}

fn report(n: i64) {
    let start = Instant::now();
    is_prime(n);
    let duration = start.elapsed();

    println!("{} is prime, elapsed time is {:?}", n, duration);
}

File: 1.22/timed_prime_test.rb
require_relative '../1.21/smallest_divisor.rb'
require_relative 'prime_test_shared.rb'

module TimedPrimeTest
  class << self
    include PrimeTestShared

    def prime?(number)
      number == SmallestDivisor.calc(number)
    end
  end
end